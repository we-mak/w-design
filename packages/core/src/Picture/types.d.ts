import { GlobProps } from "../typings";
import { WidthProps, HeightProps } from "styled-system";

type srcSetProps = {
  media: string;
  src: string;
};

export interface PictureProps extends GlobProps {
  dataSrc?: string;
  alt?: string;
  srcSets?: srcSetProps[];
  width?: WidthProps;
  height?: HeightProps;
}
