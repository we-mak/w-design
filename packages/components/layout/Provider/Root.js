"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_1 = require("../../../utils/styled");
var theme_1 = require("../../../common/theme");
var colors = theme_1.default.colors;
exports.Root = styled_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  font-family: ", ";\n\n  h1 {\n    font-size: 2rem;\n  }\n\n  h2 {\n    font-size: 1.6rem;\n  }\n\n  h3 {\n    font-size: 1.4rem;\n  }\n\n  h4 {\n    font-size: 1.2rem;\n  }\n  h5 {\n    font-size: 1rem;\n  }\n\n  h6 {\n    font-size: 0.8rem;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4,\n  h5,\n  h6 {\n    font-weight: 600;\n    line-height: 1.25;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4 {\n    color: ", ";\n  }\n\n  h5,\n  h6 {\n    color: ", ";\n  }\n\n  p {\n    color: ", ";\n    line-height: 1.5;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  a {\n    color: ", ";\n    outline: none;\n    text-decoration: none;\n\n    &:hover {\n      color: ", ";\n    }\n\n    &:focus,\n    &:hover,\n    &:active {\n      text-decoration: underline;\n    }\n    &:focus,\n    &:active {\n      color: ", ";\n    }\n    &:focus {\n      text-decoration: none;\n    }\n  }\n\n  b,\n  strong {\n    font-weight: bolder;\n  }\n\n  code,\n  kbd,\n  pre,\n  samp {\n    font-family: \"Menlo, Courier, monospace\";\n  }\n\n  code {\n    color: ", ";\n    background: ", ";\n    padding: 0.1rem 0.15rem;\n    line-height: 1.2;\n  }\n\n  small {\n    font-size: 80%;\n    font-weight: 400;\n    color: ", ";\n  }\n\n  blockquote {\n    margin-left: 0;\n    padding: 0.4rem 0.8rem;\n    color: ", ";\n    p:last-child {\n      margin-bottom: 0;\n    }\n  }\n"], ["\n  font-family: ", ";\n\n  h1 {\n    font-size: 2rem;\n  }\n\n  h2 {\n    font-size: 1.6rem;\n  }\n\n  h3 {\n    font-size: 1.4rem;\n  }\n\n  h4 {\n    font-size: 1.2rem;\n  }\n  h5 {\n    font-size: 1rem;\n  }\n\n  h6 {\n    font-size: 0.8rem;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4,\n  h5,\n  h6 {\n    font-weight: 600;\n    line-height: 1.25;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  h1,\n  h2,\n  h3,\n  h4 {\n    color: ", ";\n  }\n\n  h5,\n  h6 {\n    color: ", ";\n  }\n\n  p {\n    color: ", ";\n    line-height: 1.5;\n    margin-bottom: 0.5em;\n    margin-top: 0;\n  }\n\n  a {\n    color: ", ";\n    outline: none;\n    text-decoration: none;\n\n    &:hover {\n      color: ", ";\n    }\n\n    &:focus,\n    &:hover,\n    &:active {\n      text-decoration: underline;\n    }\n    &:focus,\n    &:active {\n      color: ", ";\n    }\n    &:focus {\n      text-decoration: none;\n    }\n  }\n\n  b,\n  strong {\n    font-weight: bolder;\n  }\n\n  code,\n  kbd,\n  pre,\n  samp {\n    font-family: \"Menlo, Courier, monospace\";\n  }\n\n  code {\n    color: ", ";\n    background: ", ";\n    padding: 0.1rem 0.15rem;\n    line-height: 1.2;\n  }\n\n  small {\n    font-size: 80%;\n    font-weight: 400;\n    color: ", ";\n  }\n\n  blockquote {\n    margin-left: 0;\n    padding: 0.4rem 0.8rem;\n    color: ", ";\n    p:last-child {\n      margin-bottom: 0;\n    }\n  }\n"])), function (props) { return props.theme.fonts; }, colors.N90, colors.N80, colors.N70, colors.B50, colors.B30, colors.B60, colors.R20, colors.N1, colors.N30, colors.N30);
exports.Root.displayName = "Root";
exports.default = exports.Root;
var templateObject_1;
//# sourceMappingURL=Root.js.map