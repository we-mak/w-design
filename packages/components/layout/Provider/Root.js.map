{"version":3,"file":"Root.js","sourceRoot":"","sources":["../../../../src/components/layout/Provider/Root.tsx"],"names":[],"mappings":";;;;;;AAAA,gDAA2C;AAC3C,+CAA0C;AAE1C,IAAM,MAAM,GAAG,eAAK,CAAC,MAAM,CAAC;AAEf,QAAA,IAAI,GAAG,gBAAM,CAAC,GAAG,k7CAAA,mBACb,EAAiC,maAyCrC,EAAU,sCAKV,EAAU,8BAIV,EAAU,oGAOV,EAAU,mFAKR,EAAU,+IAUV,EAAU,0OAoBZ,EAAU,qBACL,EAAS,mIAQd,EAAU,yFAMV,EAAU,8DAKtB,KAhHgB,UAAC,KAAU,IAAK,OAAA,KAAK,CAAC,KAAK,CAAC,KAAK,EAAjB,CAAiB,EAyCrC,MAAM,CAAC,GAAG,EAKV,MAAM,CAAC,GAAG,EAIV,MAAM,CAAC,GAAG,EAOV,MAAM,CAAC,GAAG,EAKR,MAAM,CAAC,GAAG,EAUV,MAAM,CAAC,GAAG,EAoBZ,MAAM,CAAC,GAAG,EACL,MAAM,CAAC,EAAE,EAQd,MAAM,CAAC,GAAG,EAMV,MAAM,CAAC,GAAG,EAKrB;AAEF,YAAI,CAAC,WAAW,GAAG,MAAM,CAAC;AAE1B,kBAAe,YAAI,CAAC"}