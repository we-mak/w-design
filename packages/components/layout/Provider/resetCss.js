"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var colors_1 = require("../../../common/colors");
var ResetCss = "\n*,\n*::before,\n*::after {\n  box-sizing: inherit;\n}\n\nhtml {\n  box-sizing: border-box;\n  font-size: 20px;\n  line-height: 1.5;\n  -webkit-tap-highlight-color: transparent;\n}\n\nbody {\n  background: " + colors_1.colors.WHITE + ";\n  color: " + colors_1.colors.N60 + ";\n  font-family: -apple-system, BlinkMacSystemFont, Helvetica, \"Helvetica Neue\", system-ui, sans-serif;\n  font-size: 0.8rem;\n  overflow-x: hidden;\n  text-rendering: optimizeLegibility;\n}\n\na {\n  font-family: Helvetica, \"Helvetica Neue\", system-ui, sans-serif;\n}\n\n";
exports.default = ResetCss;
//# sourceMappingURL=resetCss.js.map