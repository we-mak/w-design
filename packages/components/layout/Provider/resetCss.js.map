{"version":3,"file":"resetCss.js","sourceRoot":"","sources":["../../../../src/components/layout/Provider/resetCss.ts"],"names":[],"mappings":";;AAAA,iDAAgD;AAEhD,IAAM,QAAQ,GAAG,oNAeD,eAAM,CAAC,KAAK,oBACjB,eAAM,CAAC,GAAG,0RAWpB,CAAC;AAEF,kBAAe,QAAQ,CAAC"}