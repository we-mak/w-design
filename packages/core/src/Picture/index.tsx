/**
 * Picture
 * inherit lazy load from Image
 */
import React, { FC } from "react";
import { WidthProps, HeightProps } from "styled-system";
import Image from "../Image";
import { Source } from "./Source";
import { GlobProps } from "../common/props";

type srcSetProps = {
  media: string;
  src: string;
};

export interface PictureProps extends GlobProps {
  dataSrc?: string;
  alt?: string;
  srcSets?: srcSetProps[];
  width?: WidthProps;
  height?: HeightProps;
}

const Picture: FC<PictureProps> = ({
  srcSets,
  alt,
  dataSrc,
  width,
  height,
  ...rest
}) => {
  return (
    <picture>
      {srcSets &&
        srcSets.map((s, idx) => (
          <Source
            key={idx}
            media={`(max-width: ${s.media}px)`}
            data-srcset={s.src}
          />
        ))}
      <Image
        dataSrc={dataSrc}
        alt={alt}
        width={width}
        height={height}
        {...rest}
      />
    </picture>
  );
};

export default Picture;
