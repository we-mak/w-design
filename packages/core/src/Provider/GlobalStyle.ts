import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.medium};
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${(props: any) => props.theme.colors["N50"]};
  }

  h5,
  h6 {
    color: ${(props: any) => props.theme.colors["N60"]};
  }

  p {
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${(props: any) => props.theme.colors["N70"]};
  }

  a {
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${(props: any) => props.theme.colors["B50"]};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${(props: any) => props.theme.colors["B30"]};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${(props: any) => props.theme.colors["B60"]};
    }
    &:focus {
      text-decoration: none;
    }
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${(props: any) => props.theme.colors["N30"]};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${(props: any) => props.theme.colors["N30"]};
    p:last-child {
      margin-bottom: 0;
    }
  }

  th {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N5"]};
    padding: 0.6rem 0.4rem;
    border-bottom-width: 0.1rem;
  }
  td {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N2"]};
    padding: 0.6rem 0.4rem;
  }
`;
