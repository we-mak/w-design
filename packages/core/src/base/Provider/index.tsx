import * as React from "react";
import { ThemeProvider } from "styled-components";
import defaultTheme from "../../common/theme";
import { GlobalStyle } from "./GlobalStyle";
import "./normalize.css";

interface ThemeProps extends React.HTMLProps<HTMLDivElement> {
  theme?: object | ((theme: object) => object);
}

function Provider(p: ThemeProps) {
  const { theme = {}, children } = p;

  return (
    <ThemeProvider theme={{ ...defaultTheme, ...theme }}>
      <>
        <GlobalStyle />
        {children}
      </>
    </ThemeProvider>
  );
}

Provider.displayName = "WProvider";

export default Provider;
