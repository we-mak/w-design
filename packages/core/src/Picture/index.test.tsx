import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import Provider from "../Provider";
import Picture from "./index";

afterEach(cleanup);

describe("<Picture/>", () => {
  it("should render correct icon styled", () => {
    const { container } = render(
      <Provider>
        <Picture
          dataSrc="https://api.adorable.io/avatars/240/thunder"
          srcSets={[
            {
              media: "480",
              src:
                "https://photo2.foodgawker.com/wp-content/uploads/2019/08/3480441.jpg"
            },
            {
              media: "680",
              src:
                "https://photo2.foodgawker.com/wp-content/uploads/2019/08/3480425.jpg"
            }
          ]}
        />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
