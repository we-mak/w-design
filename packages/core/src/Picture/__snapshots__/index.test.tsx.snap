// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picture/> should render correct icon styled 1`] = `
<div>
  <picture>
    <source
      data-srcset="https://photo2.foodgawker.com/wp-content/uploads/2019/08/3480441.jpg"
      media="(max-width: 480px)"
      srcset="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
    />
    <source
      data-srcset="https://photo2.foodgawker.com/wp-content/uploads/2019/08/3480425.jpg"
      media="(max-width: 680px)"
      srcset="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
    />
    <img
      class="sc-bdfBwQ lgHXgS"
      src="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
    />
  </picture>
</div>
`;
